docs/api/subgraph/guides/v4-subgraph-example.md

```diff
@@ -0,0 +1,331 @@
+    poolCount
+    txCount
+    totalVolumeUSD
+    totalVolumeETH
+  }
+}
+```
+
+### Historical Global Data
+
+You can also query historical data by specifying a block number.
+
+```
+{
+  poolManager(
+    id: "0x000000000004444c5dc75cb358380d2e3de08a90", 
+    block: {
+      number: 22451931
+    }
+  ) {
+    poolCount
+    txCount
+    totalVolumeUSD
+    totalVolumeETH
+  }
+}
+```
+
+## Pool Data
+
+To get data about a certain pool, pass in the pool address. Reference the full [pool schema](https://github.com/Uniswap/v4-subgraph/blob/main/schema.graphql#L76) and adjust the query fields to retrieve the data points you want.
+
+### General Pool Query
+
+The query below returns the feeTier, spot price, and liquidity for the ETH-USDC pool.
+
+```
+{
+  pool(id: "0x21c67e77068de97969ba93d4aab21826d33ca12bb9f565d8496e8fda8a82ca27") {
+    tick
+    token0 {
+      symbol
+      id
+      decimals
+    }
+    token1 {
+      symbol
+      id
+      decimals
+    }
+    feeTier
+    sqrtPrice
+    liquidity
+  }
+}
+```
+
+### All Possible Pools
+
+The maximum items you can query at once is 1000. Thus to get all possible pools, you can iterate using the skip variable. To get pools beyond the first 1000 you can also set the skip as shown below.
+
+### Skipping First 1000 Pools
+
+This query sets the skip value and returns the first 10 responses after the first 1000.
+
+```
+{
+  pools(first: 10, skip: 1000) {
+    id
+    token0 {
+      id
+      symbol
+    }
+    token1 {
+      id
+      symbol
+    }
+  }
+}
+```
+
+### Creating a Skip Variable
+
+This next query sets a skip variable. In your language and environment of choice you can then iterate through a loop, query to get 1000 pools each time, and continually adjust skip by 1000 until all pool responses are returned.
+
+Check out [this example](https://github.com/Uniswap/v3-info/blob/770a05dc1a191cf229432ebc43c1f2ceb3666e3b/src/data/pools/chartData.ts#L14) from our interface for poolDayData that does something similar.
+
+> **Note**
+> This query will not work in the graph explorer and more resembles the structure of a query you'd pass to some GraphQL middleware like Apollo.
+
+```
+query pools($skip: Int!) {
+  pools(
+    first: 1000
+    skip: $skip
+    orderDirection: asc
+  ) {
+    id
+    sqrtPrice
+    token0 {
+      id
+    }
+    token1 {
+      id
+    }
+  }
+}
+```
+
+### Most Liquid Pools
+
+Retrieve the top 1000 most liquid pools. You can use this similar set up to orderBy other variables like number of swaps or volume.
+
+```
+{
+  pools(
+    first: 1000, 
+    orderBy: liquidity, 
+    orderDirection: desc
+  ) {
+    id
+  }
+}
+```
+
+### Pool Daily Aggregated
+
+This query returns daily aggregated data for the first 10 days since the given timestamp for the UNI-ETH pool. To calculate `poolId`, refer to [Note on poolId and positionId](https://docs.uniswap.org/contracts/v4/guides/state-view#note-on-poolid-and-positionid).
```

Comment:
```suggestion
This query returns daily aggregated data for the first 10 days since the given timestamp for the UNI-ETH pool. To calculate `poolId`, refer to [PoolId Library](/contracts/v4/reference/core/types/PoolId#toid).
```